ProfileManager.js

```diff
@@ -136,8 +135,15 @@ export default class ProfileManager {
     const capabilityKey = `${capabilityCacheKey}-${id}-${useEphemeralSigner}`;
 
     const capabilityCache = this._getCache(capabilityCacheKey);
-    const agentZcap = capabilityCache.get(capabilityKey);
+    const agentZcap = await capabilityCache.get(capabilityKey);
+    const now = Date.now();
     if(agentZcap) {
+      if(agentZcap.expires) {
+        const expiryDate = new Date(agentZcap.expires);
+        if(expiryDate.getTime() > now) {
+          return agentZcap;
+        }
```

Comment:
This is good except for:

1. When the zcap has expired, we need to remove it from the cache.
2. We should add some grace period where we remove the zcap from the cache if it's close to expiring. We should make that configurable -- and my first guess at a default value would be 15 minutes.